selftests/unit/test_vm.py

```diff
@@ -1,71 +1,66 @@
-        mock_vm = flexmock(_FakeVM())
-        flexmock(avocado_runner_vm).should_receive('vm_connect').and_return(mock_vm).once().ordered()
-        mock_vm.should_receive('start').and_return(True).once().ordered()
-        mock_vm.should_receive('create_snapshot').once().ordered()
-        # VMTestRunner()
-        Args = flexmock(test_result_total=1,
-                        url=['/tests/sleeptest', '/tests/other/test',
-                             'passtest'],
-                        vm_domain='domain',
-                        vm_username='username',
-                        vm_hostname='hostname',
-                        vm_port=22,
-                        vm_password='password',
-                        vm_key_file=None,
-                        vm_cleanup=True,
-                        vm_no_copy=False,
-                        vm_timeout=120,
-                        vm_hypervisor_uri='my_hypervisor_uri',
-                        env_keep=None)
-        log = flexmock()
-        log.should_receive("info")
-        job = flexmock(args=Args, log=log)
-        self.runner = avocado_runner_vm.VMTestRunner(job, None)
-        mock_vm.should_receive('stop').once().ordered()
-        mock_vm.should_receive('restore_snapshot').once().ordered()
-
-    def tearDown(self):
-        flexmock_teardown()
-
     def test_setup(self):
-        """ Tests VMTestRunner.setup() """
-        self.runner.setup()
-        self.runner.tear_down()
-        flexmock_teardown()
+        mock_vm = _FakeVM()
+        mock_vm.start = mock.Mock(return_value=True)
+        mock_vm.create_snapshot = mock.Mock()
+        mock_vm.stop = mock.Mock()
+        mock_vm.restore_snapshot = mock.Mock()
+        job_args = argparse.Namespace(test_result_total=1,
+                                      vm_domain='domain',
+                                      vm_username='username',
+                                      vm_hostname='hostname',
+                                      vm_port=22,
+                                      vm_password='password',
+                                      vm_key_file=None,
+                                      vm_cleanup=True,
+                                      vm_no_copy=False,
+                                      vm_timeout=120,
+                                      vm_hypervisor_uri='my_hypervisor_uri',
+                                      reference=['/tests/sleeptest.py',
+                                                 '/tests/other/test',
+                                                 'passtest.py'],
+                                      dry_run=True,
+                                      env_keep=None)
+        with mock.patch('avocado_runner_vm.vm_connect', return_value=mock_vm):
+            # VMTestRunner()
+            job = Job(job_args)
+            runner = avocado_runner_vm.VMTestRunner(job, None)
+            runner.setup()
+            runner.tear_down()
+        mock_vm.start.assert_called_once()
```

Comment:
I don't know what is wrong with my setup, but this always raises:

```
python selftests/unit/test_vm.py
E
======================================================================
ERROR: test_setup (__main__.VMTestRunnerSetup)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "selftests/unit/test_vm.py", line 59, in test_setup
    mock_vm.start.assert_called_once()
  File "/usr/lib/python2.7/site-packages/mock/mock.py", line 721, in __getattr__
    raise AttributeError(name)
AttributeError: assert_called_once

----------------------------------------------------------------------
Ran 1 test in 0.106s

FAILED (errors=1)
```

Reply:
Weird, I get:

```
$ python -m unittest -v selftests.unit.test_vm
test_setup (selftests.unit.test_vm.VMTestRunnerSetup) ... ok

----------------------------------------------------------------------
Ran 1 test in 0.104s

OK
```

Maybe the `mock` version being used?

```python
>>> import mock
>>> mock.__version__
'2.0.0'
```

Reply:
OK, I've checked out and this API is not available on mock 1.0.1.   The fix is:

```diff
diff --git a/selftests/unit/test_vm.py b/selftests/unit/test_vm.py
index 1f076d57..e9faddb6 100644
--- a/selftests/unit/test_vm.py
+++ b/selftests/unit/test_vm.py
@@ -57,10 +57,10 @@ class VMTestRunnerSetup(unittest.TestCase):
             runner.setup()
             runner.tear_down()
         try:
-            mock_vm.start.assert_called_once()
-            mock_vm.create_snapshot.assert_called_once()
-            mock_vm.stop.assert_called_once()
-            mock_vm.restore_snapshot.assert_called_once()
+            mock_vm.start.assert_called_once_with()
+            mock_vm.create_snapshot.assert_called_once_with()
+            mock_vm.stop.assert_called_once_with()
+            mock_vm.restore_snapshot.assert_called_once_with()
         finally:
             shutil.rmtree(job.args.base_logdir)
 
```